🛠 Tooling: Release flow missing permissions: id-token

### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/ts-api-utils/issues?q=is%3Aissue) and found none that matched my issue.

### Overview

Following #380 -> #381, it's still erroring now. https://github.com/JoshuaKGoldberg/ts-api-utils/actions/runs/7774477386/job/21231766717:

```plaintext
npm notice Publishing to https://registry.npmjs.org/ with tag latest and public access
npm ERR! code EUSAGE
npm ERR! Provenance generation in GitHub Actions requires "write" access to the "id-token" permission
```

### Additional Info

_No response_

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -39,3 +39,7 @@ on:
   push:
     branches:
       - main
+
+permissions:
+  contents: write
+  id-token: write

